[CSS Corner shape] Implement parsing for corner-shape

smfr opened the pull request:
#### 8bb1183b52ad26de532aed279fbdb9bb8e2e8471
<pre>
[CSS Corner shape] Implement parsing for corner-shape
<a href="https://bugs.webkit.org/show_bug.cgi?id=277913">https://bugs.webkit.org/show_bug.cgi?id=277913</a>
<a href="https://rdar.apple.com/133607859">rdar://133607859</a>

Reviewed by Sam Weinig.

Add support for parsing `corner-shape` and the physical and logical longhands. The &quot;edge&quot; variants are not yet supported
(similarly, we don&apos;t support the border-radius edge properties either).

Currently just support the keyword values.

Add storage of the style values in `BorderData`. Repainting happens by virtue of `isEquivalentForPainting`.

* LayoutTests/imported/w3c/web-platform-tests/css/css-borders/tentative/parsing/corner-shape-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-borders/tentative/parsing/corner-shape-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-ui/compute-kind-widget-no-fallback-props-001.html: Add corner-shape to the list of native-appearance-disabling properties.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/CSSProperty.h:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle const):
* Source/WebCore/css/ShorthandSerializer.cpp:
(WebCore::ShorthandSerializer::serialize):
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::CSSPropertyParser::parseShorthand):
* Source/WebCore/rendering/style/BorderData.cpp:
(WebCore::BorderData::dump const):
* Source/WebCore/rendering/style/BorderData.h:
(WebCore::BorderData::topLeftCornerShape const):
(WebCore::BorderData::topRightCornerShape const):
(WebCore::BorderData::bottomLeftCornerShape const):
(WebCore::BorderData::bottomRightCornerShape const):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::conservativelyCollectChangedAnimatableProperties const):
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleConstants.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/rendering/style/RenderStyleConstants.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::initialCornerShape):
(WebCore::RenderStyle::cornerBottomLeftShape const):
(WebCore::RenderStyle::cornerBottomRightShape const):
(WebCore::RenderStyle::cornerTopLeftShape const):
(WebCore::RenderStyle::cornerTopRightShape const):
* Source/WebCore/rendering/style/RenderStyleSetters.h:
(WebCore::RenderStyle::setCornerBottomLeftShape):
(WebCore::RenderStyle::setCornerBottomRightShape):
(WebCore::RenderStyle::setCornerTopLeftShape):
(WebCore::RenderStyle::setCornerTopRightShape):

Canonical link: <a href="https://commits.webkit.org/290953@main">https://commits.webkit.org/290953@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bc2d836adbc1261cde685350f6bf67088ebdac2f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/91499 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/11031 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/558 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/96468 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/42187 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/93549 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/11409 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/19429 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/70264 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/27780 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/94500 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/8708 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/82895 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/50589 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/8472 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/482 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/41357 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/84303 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/78792 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/487 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/98471 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/90248 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/18661 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/13730 "Found 1 new test failure: imported/w3c/web-platform-tests/html/browsers/the-window-object/open-close/open-features-tokenization-noreferrer.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/79292 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/18916 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/78733 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/78496 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19422 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/23010 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/370 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/11789 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/18659 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/23935 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/112832 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/18369 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/32715 "Found 11 new JSC stress test failures: microbenchmarks/memcpy-wasm-medium.js.bytecode-cache, microbenchmarks/memcpy-wasm-medium.js.default, microbenchmarks/memcpy-wasm-medium.js.no-llint, microbenchmarks/memcpy-wasm-small.js.no-llint, wasm.yaml/wasm/stress/b3-signed-extend-16-to-64.js.wasm-collect-continuously, wasm.yaml/wasm/stress/ipint-bbq-osr-with-try5.js.wasm-eager-jettison, wasm.yaml/wasm/stress/live-funcref-across-loop-tier-up.js.default-wasm, wasm.yaml/wasm/stress/live-funcref-across-loop-tier-up.js.wasm-collect-continuously, wasm.yaml/wasm/stress/repro_1289.js.default-wasm, wasm.yaml/wasm/stress/repro_1289.js.wasm-slow-memory ... (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/21829 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/20135 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/CSSProperties.json:
```diff
@@ -3821,6 +3821,169 @@
+                    "name": "corner-shape",
+                    "resolver": "top-left"
+                },
+                "parser-grammar": "round | scoop | bevel | notch | straight",
```
```suggestion
                "parser-grammar": "<corner-shape-value>",
```

You can just refer to `<corner-shape-value>` directly, ditto below

smfr committed:
[CSS Corner shape] Implement parsing for corner-shape
https://bugs.webkit.org/show_bug.cgi?id=277913
rdar://133607859

Reviewed by Sam Weinig.

Add support for parsing `corner-shape` and the physical and logical longhands. The "edge" variants are not yet supported
(similarly, we don't support the border-radius edge properties either).

Currently just support the keyword values.

Add storage of the style values in `BorderData`. Repainting happens by virtue of `isEquivalentForPainting`.

* LayoutTests/imported/w3c/web-platform-tests/css/css-borders/tentative/parsing/corner-shape-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-borders/tentative/parsing/corner-shape-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-ui/compute-kind-widget-no-fallback-props-001.html: Add corner-shape to the list of native-appearance-disabling properties.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/CSSProperty.h:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle const):
* Source/WebCore/css/ShorthandSerializer.cpp:
(WebCore::ShorthandSerializer::serialize):
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::CSSPropertyParser::parseShorthand):
* Source/WebCore/rendering/style/BorderData.cpp:
(WebCore::BorderData::dump const):
* Source/WebCore/rendering/style/BorderData.h:
(WebCore::BorderData::topLeftCornerShape const):
(WebCore::BorderData::topRightCornerShape const):
(WebCore::BorderData::bottomLeftCornerShape const):
(WebCore::BorderData::bottomRightCornerShape const):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::conservativelyCollectChangedAnimatableProperties const):
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleConstants.cpp:
(WebCore::operator<<):
* Source/WebCore/rendering/style/RenderStyleConstants.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::initialCornerShape):
(WebCore::RenderStyle::cornerBottomLeftShape const):
(WebCore::RenderStyle::cornerBottomRightShape const):
(WebCore::RenderStyle::cornerTopLeftShape const):
(WebCore::RenderStyle::cornerTopRightShape const):
* Source/WebCore/rendering/style/RenderStyleSetters.h:
(WebCore::RenderStyle::setCornerBottomLeftShape):
(WebCore::RenderStyle::setCornerBottomRightShape):
(WebCore::RenderStyle::setCornerTopLeftShape):
(WebCore::RenderStyle::setCornerTopRightShape):

Canonical link: https://commits.webkit.org/290953@main
